### PR TITLE
partners: Add Hybrid Search to MongoDB's VectorStore

### PR DESCRIPTION
**Pull request by caseyclements**
## Description

This pull-request extends the existing vector search strategies of MongoDBAtlasVectorSearch (currently HNSW) to include Hybrid (Reciprocal Rank Fusion) and Full-text. 

### Checklist
* Unit tests have been extended
    * One with Chat Model closely follows https://python.langchain.com/v0.2/docs/how_to/hybrid/
* formatting has been applied
* On linting, some docstrings run over 88 characters. Code looks compliant.

### CODE REVIEW

File: libs/partners/mongodb/langchain_mongodb/pipelines.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+def text_search_stage(
+    query: str, search_field, index_name: str, operator: str = "phrase", **kwargs: Any
+) -> Dict[str, Any]:
+    """Full-Text search.
+
+    Args:
+        query: Input text to search for
+        search_field: Field in Collection that will be searched
+        index_name: Atlas Search Index name
+        operator: A number of operators are available in the text search stage.
```
Comment (Jibola) [inline libs/partners/mongodb/langchain_mongodb/pipelines.py:18]:
```suggestion
        operator: operator available in the $search stage.
```

**Comment by efriis** [comment]
Going to close this draft as stale. Feel free to reopen if you pick it up again!

**Comment by caseyclements** [comment]
Hi @efriis. All good. I was at SciPy last week. I'll reopen soon. I am going to refactor into a core VectorStore and a Retriever for Hybrid search.

**Comment by sivakumar41** [comment]
> Hi @efriis. All good. I was at SciPy last week. I'll reopen soon. I am going to refactor into a core VectorStore and a Retriever for Hybrid search.

Can we use the hybrid search of the mongodb?

**Comment by prakul** [comment]
@sivakumar41 This functionality is being worked on and you cant use it yet

**Comment by sivakumar41** [comment]
> @sivakumar41 This functionality is being worked on and you cant use it yet

Can you share the reference

**Pull request closed without merging**